orttraining/orttraining/python/training/torchdynamo/ort_backend.py

```diff
@@ -0,0 +1,474 @@
+            node.target = node.target.overloadpacket
+    fx_module.graph.lint()
+    fx_module.recompile()
+    return torch.jit.script(fx_module)
+
+
+def _decorate_script_module(script_module: torch.jit.ScriptModule, expected_inputs, expected_outputs):
+    for i, input_value in enumerate(script_module.graph.inputs()):
+        if input_value.debugName() == "self":
+            script_module.graph.eraseInput(i)
+            break
+    for i, input_value in enumerate(script_module.graph.inputs()):
+        input_value.setType(torch._C.TensorType.create_from_tensor(expected_inputs[i]))
+    for i, output_value in enumerate(script_module.graph.outputs()):
+        output_value.setType(torch._C.TensorType.create_from_tensor(expected_outputs[i]))
+
+
+def _create_onnx_proto(script_module):
+    onnx_proto = _jit_graph_to_onnx_model(script_module.graph, torch.onnx.OperatorExportTypes.ONNX)
+    return onnx_proto
+
+
+def _create_onnx_model(onnx_proto):
+    return onnx.ModelProto.FromString(onnx_proto)
+
+
+def _create_onnx_session(onnx_proto, device):
+    # TODO(wechi): Add more EPs per PyTorch device types.
+    # TODO(wechi): enable external allocators.
+    if device.type == "cuda":
+        return onnxruntime.InferenceSession(onnx_proto, providers=["CUDAExecutionProvider"])
+    return onnxruntime.InferenceSession(onnx_proto, providers=["CPUExecutionProvider"])
+
+
+def _get_onnx_devices(values: Tuple[torch.Tensor]) -> List[ORTC.OrtDevice]:
+    devices: List[ORTC.OrtDevice] = []
+    first_device = values[0].device
+    for value in values:
+        assert value.device == first_device, "All values must be on the same device"
+        device_id = value.device.index or 0
+        devices.append(
+            ORTC.OrtDevice(
+                _get_ort_device_type(value.device.type, device_id), ORTC.OrtDevice.default_memory(), device_id
+            )
+        )
+    return devices
+
+
+def _run_onnx_session_with_ortvaluevector(
+    sess: onnxruntime.InferenceSession,
+    input_names: List[str],
+    inputs: Tuple[torch.Tensor],
+    input_devices: List[ORTC.OrtDevice],
+    output_names: List[str],
+    outputs: Tuple[torch.Tensor],
+    output_devices: List[ORTC.OrtDevice],
+):
+    _nvtx_range_push("contiguous")
+    inputs = tuple(a.contiguous() for a in inputs)
+    _nvtx_range_pop()
+
+    _nvtx_range_push("push_back_batch")
+    ort_inputs = ORTC.OrtValueVector()
+    ort_inputs.reserve(len(inputs))
+    ort_outputs = ORTC.OrtValueVector()
+    dtypes = []
+    shapes = []
+    data_ptrs = []
+
+    for value in inputs:
+        dtypes.append(_NP_DTYPE[value.dtype])
+        shapes.append(value.size())
+        data_ptrs.append(value.data_ptr())
+    ort_inputs.push_back_batch(inputs, data_ptrs, dtypes, shapes, input_devices)
+    _nvtx_range_pop()
+
+    _nvtx_range_push("run_with_ortvaluevector")
+    run_options = onnxruntime.RunOptions()
+    run_options.synchronize_execution_providers = False
```

Comment:
Default here is `False`, but on https://github.com/microsoft/onnxruntime/pull/13259/files#diff-a57e626ba456c2386c8a00eabf477ec075027fc12f0a6054e46c594cdef687baR32 is `true`;

Is this intentional?

Reply:
This was intentional but I feel I was wrong. Let me change this back to `False` in this PR and set it to `True` in my experiment script.